.evergreen/scripts/setfle2parameter.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+"""
+setfle2parameter.py modifies and prints an orchestration config file to add the `--setParameter featureFlagFLE2ProtocolVersion2=1` option to mongod and mongos.
+Usage: setfle2parameter.py <orchestration config file path>
+This file is a temporary workaround until featureFlagFLE2ProtocolVersion2 is enabled by default in SERVER-69563. Once latest server builds have SERVER-69563, this file may be removed.
+DRIVERS-2590 tracks removal of this file.
+"""
+
+import json
+import sys
+import unittest
+import os
+
+
+def do_rewrite(config):
+    did_rewrite = False
+
+    def rewrite_server(server):
+        if "procParams" in server:
+            if "setParameter" in server["procParams"]:
+                server["procParams"]["setParameter"]["featureFlagFLE2ProtocolVersion2"] = "1"
+            else:
+                server["procParams"]["setParameter"] = {
+                    "featureFlagFLE2ProtocolVersion2": "1"
+                }
+            return True
+        return False
+
+    # Rewrite for a server.
+    if rewrite_server(config):
+        did_rewrite = True
+    # Rewrite for each member in a replica set.
+    if "members" in config:
+        for server in config["members"]:
+            if rewrite_server(server):
+                did_rewrite = True
+    # Rewrite each shard.
+    if "shards" in config:
+        for shard in config["shards"]:
+            if "shardParams" in shard:
+                if "members" in shard["shardParams"]:
+                    for server in shard["shardParams"]["members"]:
+                        if rewrite_server (server):
+                            did_rewrite = True
+    # Rewrite each router.
+    if "routers" in config:
+        for router in config["routers"]:
+            # routers do not use `procParams`. Use setParameter directly.
+            if "setParameter" in router:
+                router["setParameter"]["featureFlagFLE2ProtocolVersion2"] = "1"
+            else:
+                router["setParameter"] = {
+                    "featureFlagFLE2ProtocolVersion2": "1"
+                }
+            did_rewrite = True
```

Comment:
I got nerd-sniped into flattening this code :)

```suggestion
    import itertools

    # We will rewrite the root config:
    root = [config]
    # And any top-level members:
    root_members = config.get("members", [])

    # As well as the members within any defined shards:
    shards = config.get("shards", [])
    # Get a list of lists of membres:
    shard_member_lists = (s.get("shardParams", {}).get("members", []) for s in shards)
    # Flatten the list of lists:
    shard_members = itertools.chain.from_iterable(shard_member_lists)

    # For all members in all groups:
    all_members = itertools.chain(root, root_members, shard_members)
    # Rewrite them all:
    all_rewrites = list(map(rewrite_server, all_members))
    # Did we actually rewrite anything?
    did_rewrite = any(all_rewrites)

    # Rewrite each router.
    for rtr in config.get("routers", []):
        ps = rtr.setdefault("setParameter", {})
        ps["featureFlagFLE2ProtocolVersion2"] = "1"
        did_rewrite = True
```

[`itertools.chain` reference](https://docs.python.org/3/library/itertools.html#itertools.chain)

Reply:
Oh that is really neat. I did not know about `itertools.chain`.